python_modules/libraries/dagster-elt/dagster_elt/resources.py

```diff
@@ -0,0 +1,189 @@
+import contextlib
+import os
+import tempfile
+from enum import Enum
+from subprocess import PIPE, STDOUT, Popen
+from typing import Any, Dict, Generator, List, Optional
+
+import yaml
+from dagster import Config, ConfigurableResource, get_dagster_logger
+from dagster._utils.env import environ
+from pydantic import Field
+from sling import Sling  # type: ignore
+
+logger = get_dagster_logger()
+
+
+class SlingMode(str, Enum):
+    """The mode to use when syncing.
+    See the Sling docs for more information: https://docs.slingdata.io/sling-cli/running-tasks#modes.
+    """
+
+    INCREMENTAL = "incremental"
+    TRUNCATE = "truncate"
+    FULL_REFRESH = "full-refresh"
+    SNAPSHOT = "snapshot"
+
+
+class SlingSource(Config):
+    """Sling Source configuration.
+
+
+    See the docs for more information:
+    https://docs.slingdata.io/sling-cli/running-tasks#cli-flags-yaml-keys
+
+    """
+
+    stream: Optional[str] = Field(
+        description=(
+            "The source table (schema.table) or local / cloud file path. Can also be the path of"
+            " sql file or in-line text."
+        )
+    )
+    primary_key: Optional[List[str]] = Field(description="The column(s) to use as primary key.")
+    update_key: Optional[List[str]] = Field(
+        description="The column(s) to use as update key. If not provided, primary key will be used."
+    )
+    limit: Optional[int] = Field(
+        description=(
+            "The number of rows to fetch from the source. If not provided, all rows will be "
+            "fetched."
+        )
+    )
+
+    options: Optional[Dict[str, Any]] = Field(
+        description="Additional options to pass to the source. See Sling docs for more details."
+    )
+
+
+class SlingTarget(Config):
+    object: str = Field(
+        description=(
+            "The target table (schema.table) or local / cloud file path. Can also be the path of"
+        )
+    )
+
+
+class SlingResource(ConfigurableResource):
+    """Resource for interacting with the Sling package.
+
+    Examples:
+        .. code-block:: python
+        from dagster import Definitions, asset
+        from dagster_etl import SlingResource
+
+        @asset
+        def sync_asset(sling: SlingResource):
```

Comment:
This could lead people to think that defining your own asset is more common than using `build_sling_asset`. Is that the case?